[Sheryl-Lynn Tan] iP

sltsheryl opened the pull request:
### Kirby Deluxe Bot
> “By failing to prepare, you are preparing to fail.” – Benjamin Franklin, _Founding Father of the United States_

The once-in-a-lifetime Kirby Deluxe Bot provides you unprecedented ease in managing your tasks and helps you achieve maximum productivity as soon as you use it! 🤯

It's text-based, speedy, easy to use and BRIGHTENS UP YOUR DAY :D 

All you need to do is download it from [here](https://github.com/sltsheryl/ip/releases/tag/A-JAR). After downloading the jar file, 
1. Open a command window in that folder
2. Run the command `java -jar kirby.jar` (i.e., run the command in the same folder as the jar file)

Features:
- [x] Managing tasks
- [x] Managing deadlines 
- [x] Managing events

Look out for more features soon!! 🤠

Commit:
Add Gradle support

sltsheryl committed:
Greet, Echo, Exit

sltsheryl committed:
Change name

sltsheryl committed:
Add Level

sltsheryl committed:
Add mark as done

sltsheryl committed:
Todo, event, deadline

sltsheryl committed:
Add methods

sltsheryl committed:
A-TextUiTesting

sltsheryl committed:
Add one more test case

sltsheryl committed:
Update letter casing

sltsheryl committed:
Handle Exceptions

sltsheryl committed:
Delete

sltsheryl committed:
Add Javadocs

sltsheryl committed:
Add Enum

sltsheryl committed:
Correct switch indentation

sltsheryl committed:
Add save feature & Modify access modifiers

sltsheryl committed:
Change access modifiers

sltsheryl committed:
Add Dates and Times

sltsheryl committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Kirby.java

sltsheryl committed:
Remove data folder

sltsheryl committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Kirby.java
#	src/main/java/Todo.java

sltsheryl committed:
Reformat packages and comments

sltsheryl committed:
Add OOP Classes

sltsheryl committed:
Add new classes

sltsheryl committed:
Add packages

sltsheryl committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Add gradle from remote repo

sltsheryl committed:
Add JUnit tests

sltsheryl committed:
Edit unit test files

sltsheryl committed:
Add gradle dependencies

sltsheryl committed:
Add Javadocs

sltsheryl committed:
Edit formatting to follow Coding Standard

sltsheryl committed:
Add Find command

sltsheryl committed:
Minor edit to JavaDocs

sltsheryl committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Minor edit to JavaDocs

sltsheryl committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit formatting to follow Coding Standard

# Conflicts:
#	src/main/java/kirby/commands/Command.java

sltsheryl committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Find command

# Conflicts:
#	src/main/java/kirby/TaskList.java
#	src/main/java/kirby/tasks/Task.java

sltsheryl committed:
Add Javadocs for FindCommand

fannyjian commented on src/main/java/kirby/HandleTime.java:
```diff
@@ -0,0 +1,64 @@
+     * @param input the string input from the user.
+     * @return an array of the date in the form of [dd, mm, yyyy].
+     */
+    public static int[] fromStringToDate(String input) {
```
I like the way you order your method names sensibly!

fannyjian commented on src/main/java/kirby/HandleTime.java:
```diff
@@ -0,0 +1,64 @@
+     * @param inDate the input to be checked if it is a date.
+     * @return true if it is a date or false if it is not in standard form of a date.
+     */
+    public static boolean isValidDate(String inDate) {
```
Good name for a method that returns a boolean!

fannyjian commented on src/test/java/kirby/HandleTimeTest.java:
```diff
@@ -0,0 +1,31 @@
+
+public class HandleTimeTest {
+    @Test
+    public void negativeCheckDate() {
```
Perhaps you could use a different word instead of "negative" to describe this test? I found it a bit confusing to understand the function from the name alone.

malwaregarry replied:
Since you are testing `isValidDate()` perhaps naming it like this would be more clear?
```suggestion
    public void isValidDate_false() {
```

sltsheryl replied:
Thanks :)

fannyjian reviewed (commented):
Generally, I think your method and variable names are consistent with the coding standards!

malwaregarry commented on src/main/java/kirby/Parser.java:
```diff
@@ -0,0 +1,53 @@
+package kirby;
+
+import kirby.commands.*;
```
Perhaps you could list the individual commands? While it might look cleaner, I believe wildcard imports are not allowed by the coding standard.

sltsheryl replied:
Ah yes, it pops up on Checkstyle too, amended. Thanks!

malwaregarry commented on src/main/java/kirby/Parser.java:
```diff
@@ -0,0 +1,53 @@
+     * @throws KirbyInvalidCommandException if the command entered is undefined.
+     */
+    public static Command parse(String inputString) throws KirbyInvalidCommandException {
+        switch (inputString.split(" ")[0]) {
```
Great that you followed the switch-case format as stated in the coding standard. Did you edit the IDE settings for this or manually un-indented it yourself?

sltsheryl replied:
I edited the IDE settings

malwaregarry commented on src/main/java/kirby/Kirby.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * Constructor for the class Kirby.
+     *
+     * @param filePath name of the file path as a string.
```
Not too sure myself but I believe the first letter of of the `param`, `return` and `throws` statements should be captitalised
```suggestion
     * @param filePath Name of the file path as a string.
```

sltsheryl replied:
You're right thanks!

malwaregarry commented on src/main/java/kirby/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * {@inheritDoc}
+     * Creates a Deadline task if arguments are valid.
+     */
```
I like that you provided comments for all public methods. Good news is that prof damith has recently removed the requirement of needing javadoc comments for overriding methods.
https://se-education.org/guides/conventions/java/intermediate.html
> Write descriptive header comments for all public classes/methods.
>
>You MUST write header comments for all classes, public methods. But they can be omitted for the following cases:
>i. Getters/setters
>ii. When overriding methods (provided the parent method's Javadoc applies exactly as is to the overridden method)

malwaregarry commented on src/main/java/kirby/exceptions/KirbyInvalidCommandException.java:
```diff
@@ -0,0 +1,15 @@
+     * Constructor for the class KirbyInvalidCommandException.
+     */
+    public KirbyInvalidCommandException() {
+        super("Aaaaaaah, I don't understand you :(\n" +
+                "Try adding a valid command (todo, deadline, event)");
```
Not sure if this is required as it is not explicitly stated, but do you think it would be better if the `+` is at the start of the next line instead of end of the previous line?
```suggestion
        super("Aaaaaaah, I don't understand you :(\n"
                + "Try adding a valid command (todo, deadline, event)");
```

sltsheryl replied:
Not sure, also can't find any rule about it. Personally, I find the '+' above easier to read but if there's a more correct way, I'll follow it.

Update: Checkstyle says everything in one line 😅

malwaregarry commented on src/test/java/kirby/TaskListTest.java:
```diff
@@ -0,0 +1,33 @@
+
+public class TaskListTest {
+    @Test
+    public void createTaskList() {
```
Great work on the extensive testing

malwaregarry reviewed (commented):
Overall, I liked how your program is structured and the personality you gave to the assistant. 
It was quite difficult to find any coding violations in your code and the comments were really nitpicks.

sltsheryl committed:
Modify to improve aligning to coding standard from peer feedback

sltsheryl committed:
Change based on Checkstyle

sltsheryl commented:
> Overall, I liked how your program is structured and the personality you gave to the assistant. It was quite difficult to find any coding violations in your code and the comments were really nitpicks.

Thanks for your detailed feedback too!

sltsheryl committed:
Add JavaFX GUI

sltsheryl committed:
Customise background and GUI style

sltsheryl committed:
Edit GUI fonts

sltsheryl committed:
Fix delete task bug

sltsheryl committed:
Delete unnecessary images

jgyj123 commented on src/main/java/kirby/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+     *
+     * @param taskNumber Index of the task to be marked.
+     */
+    public void setTaskMarked(int taskNumber) {
```
Really good naming of setters - I immediately know what the method is used for 👍

sltsheryl committed:
Amend JavaDocs

jgyj123 commented on src/main/java/kirby/tasks/Task.java:
```diff
@@ -0,0 +1,67 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Great that you have followed the following coding standard - Boolean variables/methods should be named to sound like booleans. Well done!

jgyj123 commented on src/test/java/kirby/HandleTimeTest.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Test
+    public void isValidDate_true() {
```
Good naming of test methods to show the expected result. Well done!

jgyj123 reviewed (commented):
Overall, the coding standard was extremely well followed. Could not really find any violations in the code. LGTM.

Well Done! 👍

sltsheryl committed:
Minor edit to GUI (position and padding)

sltsheryl committed:
Add text bubbles

sltsheryl committed:
Fix wrap text issues

sltsheryl committed:
Create overdue level 10 branch

sltsheryl committed:
Revert "Create overdue level 10 branch"

This reverts commit 6baaa1393981d60b6190f3743366ab754dd886c8.

sltsheryl committed:
Add message to GUI

sltsheryl committed:
Merge branch 'branch-Level-10'
Merge overdue Level 10 branch to main.

sltsheryl committed:
Modify wrap text

sltsheryl committed:
Modify shadowJar class

sltsheryl committed:
Add assertions

* Assertions are used to ensure several method arguments are non-null
* Assertion is then enabled on Intellij

sltsheryl committed:
Refactor to improve code quality
Following aspects were adhered to:
* Avoid magic numbers and strings - create static variables with easy-to-follow names
* Rename resultant list to describe the output more clearly
* Avoid long methods - introduce boolean variables for long checks

sltsheryl committed:
Set up Continuous Integration (CI)
Its purpose is so that GitHub Actions will run the workflow (as per the .yml file) every time certain project events are triggered (e.g., when a PR is updated, or the master branch is updated).

sltsheryl committed:
Merge pull request #1 from sltsheryl/branch-A-Assertions

Add assertions

sltsheryl committed:
Merge branch 'master' into branch-A-CodeQuality

sltsheryl committed:
Merge pull request #2 from sltsheryl/branch-A-CodeQuality

Refactor to improve code quality

sltsheryl committed:
Merge pull request #3 from sltsheryl/branch-A-CI

Set up Continuous Integration (CI)

sltsheryl committed:
Add sort by date feature

sltsheryl committed:
Add JavaDocs

sltsheryl committed:
Merge branch 'branch-C-Sort'

* branch-C-Sort:
  Add JavaDocs
  Add sort by date feature

sltsheryl committed:
Fix ShowList bug
Incorrectly added append(1)

sltsheryl committed:
Add Help feature
The Help feature displays all the available commands.

sltsheryl committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
  Add Help feature The Help feature displays all the available commands.

sltsheryl committed:
Add and tidy up JavaDocs

sltsheryl committed:
Fix Set Up CI to correct directory

sltsheryl committed:
Fix checkstyle

sltsheryl committed:
Fix checkstyle for test files

sltsheryl committed:
Fix test cases bugs

sltsheryl committed:
Add find command in HelpCommand

sltsheryl committed:
Refactor code to improve design
* Handle argument checks in Parser
* Handle exceptions more elegantly
* Reduce repetition of code for printing of list
* Rename exceptions more clearly

sltsheryl committed:
Fix unit test case

sltsheryl committed:
Fix logic for ExitCommand and HelpCommand

sltsheryl committed:
Refactor to improve GUI
* Resize profile picture
* Edit shape of chat bubbles
* Change dialogue boxes color

sltsheryl committed:
Add Parser unit testing

sltsheryl committed:
Refactor retrieve arguments function for Parser
Handle invalid inputs for Deadline, Event and Todo

sltsheryl committed:
Modify error message and corresponding unit test output

sltsheryl committed:
Update README.md

sltsheryl committed:
Enlarge README.md font size

sltsheryl committed:
Create User Guide

sltsheryl committed:
Add Ui picture file

sltsheryl committed:
Add pictures for better visuals

sltsheryl committed:
Fix table on Github Pages

sltsheryl committed:
Add User Guide links

sltsheryl committed:
Fix creation of Jar file in build.gradle

sltsheryl committed:
Add jar file

sltsheryl committed:
Update README.md

sltsheryl committed:
Update phrasing in README.md

sltsheryl committed:
Correct grammatical error in README.md

sltsheryl committed:
Add list command message

sltsheryl committed:
Update jar and edit typo

sltsheryl committed:
Add resizing GUI and references

sltsheryl committed:
Fix reference typo

sltsheryl committed:
Add font library

sltsheryl committed:
Fix checkstyle error

sltsheryl committed:
Enlarge fonts

sltsheryl committed:
Fix Mark when storing file
